Build issue when NNG_ENABLE_STATS is OFF

## NNG & Platform details.

- version: 1.2.5
- platform: linux, macos

## Expected Behavior

When `NNG_ENABLE_STATS:BOOL=OFF`, the `make` will fail as follows:

```
[ 49%] Building C object src/CMakeFiles/nng.dir/core/protocol.c.o
[ 50%] Building C object src/CMakeFiles/nng.dir/core/random.c.o
[ 52%] Building C object src/CMakeFiles/nng.dir/core/reap.c.o
[ 53%] Building C object src/CMakeFiles/nng.dir/core/socket.c.o
[ 55%] Building C object src/CMakeFiles/nng.dir/core/stats.c.o
/tmp/nng-1.2.5/src/core/stats.c: In function ‘nni_stat_register’:
/tmp/nng-1.2.5/src/core/stats.c:63:16: error: ‘stats_lock’ undeclared (first use in this function)
  nni_mtx_lock(&stats_lock);
                ^
/tmp/nng-1.2.5/src/core/stats.c:63:16: note: each undeclared identifier is reported only once for each function it appears in
/tmp/nng-1.2.5/src/core/stats.c:64:16: error: ‘stats_root’ undeclared (first use in this function)
  nni_stat_add(&stats_root, child);
                ^
src/CMakeFiles/nng.dir/build.make:543: recipe for target 'src/CMakeFiles/nng.dir/core/stats.c.o' failed
make[2]: *** [src/CMakeFiles/nng.dir/core/stats.c.o] Error 1
CMakeFiles/Makefile2:603: recipe for target 'src/CMakeFiles/nng.dir/all' failed
make[1]: *** [src/CMakeFiles/nng.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2
```

## Actual Behavior

The error line is here: https://github.com/nanomsg/nng/blob/v1.2.5/src/core/stats.c#L63

Function `nni_stat_register` is built always independent of the `NNG_ENABLE_STATS` config, while the variable `stats_lock` referenced in it is only defined when the config is enabled at https://github.com/nanomsg/nng/blob/v1.2.5/src/core/stats.c#L34

## Steps to Reproduce

```shell
mkdir build
cd build
cmake -DNNG_ENABLE_STATS:BOOL=OFF ..
make
```

--- a/src/core/stats.c
+++ b/src/core/stats.c
@@ -60,9 +60,13 @@ nni_stat_add(nni_stat_item *parent, nni_stat_item *child)
 void
 nni_stat_register(nni_stat_item *child)
 {
+#ifdef NNG_ENABLE_STATS
 	nni_mtx_lock(&stats_lock);
 	nni_stat_add(&stats_root, child);
 	nni_mtx_unlock(&stats_lock);
+#else
+	NNI_ARG_UNUSED(child);
+#endif
 }
 
 void

